openquake/hazardlib/gsim/abrahamson_2014.py

```diff
@@ -0,0 +1,520 @@
+        fake_z1pt0 = np.ones_like(sites.vs30) * -1
+        return (self._get_basic_term(C, rup, dists) +
+                self._get_faulting_style_term(C, rup) +
+                self._get_site_response_term(C, imt, vs30_1180, ref_iml) +
+                self._get_hanging_wall_term(C, dists, rup) +
+                self._get_top_of_rupture_depth_term(C, rup) +
+                self._get_soil_depth_term(C, fake_z1pt0, vs30_1180) +
+                self._get_regional_term(C, vs30_1180, imt, dists.rrup)
+                )
+
+    def _get_basic_term(self, C, rup, dists):
+        """
+        Compute and return basic form, see page 1030.
+        # Tested for m <= 4.9
+        """
+        # Fictitious depth calculation
+        if rup.mag > 5.:
+            c4m = C['c4']
+        elif rup.mag > 4.:
+            c4m = C['c4'] - (C['c4']-1.) * (5. - rup.mag)
+        else:
+            c4m = 1.
+        R = np.sqrt(dists.rrup**2. + c4m**2.)
+
+        # basic form
+        base_term = C['a1'] * np.ones_like(dists.rrup) + C['a17'] * dists.rrup
+        # equation 2 at page 1030
+        if rup.mag >= C['m1']:
+            base_term += (C['a5'] * (rup.mag - C['m1']) +
+                          C['a8'] * (8.5 - rup.mag)**2. +
+                          (C['a2'] + C['a3'] * (rup.mag - C['m1'])) *
+                          np.log(R))
+        elif rup.mag >= self.CONSTS['m2']:
+            base_term += (C['a4'] * (rup.mag - C['m1']) +
+                          C['a8'] * (8.5 - rup.mag)**2. +
+                          (C['a2'] + C['a3'] * (rup.mag - C['m1'])) *
+                          np.log(R))
+        else:
+            base_term += (C['a4'] * (self.CONSTS['m2'] - C['m1']) +
+                          C['a8'] * (8.5 - self.CONSTS['m2'])**2. +
+                          C['a6'] * (rup.mag - self.CONSTS['m2']) +
+                          C['a7'] * (rup.mag - self.CONSTS['m2'])**2. +
+                          (C['a2'] + C['a3'] * (self.CONSTS['m2'] - C['m1'])) *
+                          np.log(R))
+        return base_term
+
+    def _get_faulting_style_term(self, C, rup):
+        """
+        Compute and return faulting style term, that is the sum of the second
+        and third terms in equation 1, page 74.
+        """
+        # this implements equations 5 and 6 at page 1032. f7 is the
+        # coefficient for reverse mechanisms while f8 is the the correction
+        # factor for normal ruptures
+        if rup.mag > 5.0:
+            f7 = C['a11']
+            f8 = C['a12']
+        elif rup.mag >= 4:
+            f7 = C['a11'] * (rup.mag - 4.)
+            f8 = C['a12'] * (rup.mag - 4.)
+        else:
+            f7 = 0.0
+            f8 = 0.0
+        # ranges of rake values for each faulting mechanism are specified in
+        # table 2, page 1031
+        return (f7 * float(rup.rake > 30 and rup.rake < 150) +
+                f8 * float(rup.rake > -150 and rup.rake < -30))
+
+    def _get_vs30star(self, vs30, imt):
+        """
```

Comment:
The lack of comment inside the comment string is usually a PEP8 violation. Is there anything that can be put here to explain the function?